docs/proposals/hpa/centralized-hpa.md

```diff
@@ -0,0 +1,183 @@
+
+#### Story 4
+As a user, I deploy my application with HPA in multiple clusters. Since the unlimited scaling up of the application may lead to unexpected cloud costs and resource congestion that may cause service disruptions for other applications, so I expect to set the maximum number of instances for multi-cluster applications.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go into as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+1. The workloads/pods in different member clusters selected by the same HPA CR/resource share the load of the application equally. For example, 10 pods of the application are spread into two member clusters with distribution `cluster1: 3 pods, cluster2: 7 pods`, so the 3 pods in cluster1 take 3/10 of total requests and 7 pods in cluster2 take 7/10 of total requests. Scenarios that don't meet the restriction are not considered in this proposal.
+1. If the system does not meet the first requirement, some clusters' services may become overloaded while others remain underloaded.      
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate them? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. If the HPA is propagated to member clusters, there will be conflicts with Karmada control plane. So users couldn't propagate HPA from Karmada control plane.
+2. The HPA used in a single cluster can be seamlessly migrated to multiple clusters' centralized HPA.
+
+## Design Details
+### Architecture
+For different scenarios, the architecture may be different:  
+HPA with CPU/Memory metrics:  
+![K8s native metrics](./statics/centralizedhpa-native-metrics.png)  
+
+HPA with custom metrics:  
+![Custom metrics](./statics/centralizedhpa-custom-metrics.png)  
+
+HPA with managed metrics store:  
+![Managed metrics store](./statics/managed-metrics-store.png)  
+
+From the above architecture, we can get following information:
+* Centralized HPA controller is the original HPA controller in `karmada-kube-controller-manager` component, which will query metrics from `karmada-apiserver` with metrics API and scale the deployments.
+* `karmada-metrics-adaptor` is responsible for query metrics from member clusters and do aggregation.
+* Centralized HPA only responsible for scaling instances but doesn't care where the instances will be scheduled.
+* The HPA version should be compatible with `karmada-kube-controller-manager` version.
+
+For better user experience, centralized HPA API is same with original HPA API.
+
+### Components change
+
+#### karmada-kube-controller-manager  
+In order to make HPA scaling work, HPA controller should be enablded in `karmada-kube-controller-manager` as follows:
```

Comment:
We may introduce a new API to distinguish kubernetes native hpa and      
the multi-cluster hpa. So we might not use the original API though they may have the same fields.